byterun/caml/memory.h

```diff
@@ -465,6 +464,8 @@ CAMLextern struct caml__roots_block *caml_local_roots;  /* defined in roots.c */
   caml_modify (&Field ((block), caml__temp_offset), caml__temp_val); \
 }while(0)
 
+#if CAML_API_VERSION < 400
+
 /*
```

Comment:
I'm afraid that `Begin_roots` and `End_roots` are used in code generated by the venerable CamlIDL tool (https://github.com/xavierleroy/camlidl/).  The tool hasn't been updated in a long time but still has users.  Forcing them to set `CAML_API_VERSION` to 312 or something is not nice.

Reply:
Okay. Now that this is on `trunk` and not `4.07` I'll get around to this after my vacation.

Reply:
So, first question about this: if the venerable *CamlIDL* doesn't set `CAML_API_VERSION` then it defaults to 100, and `Begin_roots`/`End_roots` definitions are available accordingly. Am I mistaken?